presets/ragengine/main.py

```diff
@@ -253,6 +253,83 @@ async def list_documents_in_index(
     except Exception as e:
         raise HTTPException(status_code=500, detail=str(e))
 
+@app.post(
+    "/indexes/{index_name}/documents/{doc_id}",
+    response_model=DocumentResponse,
+    summary="Update document in an Index by its ID",
+    description="""
+    Update document in an Index by its ID.
+
+    ## Request Example:
+    ```json
+    POST /indexes/test_index/documents/test_doc_id
+    {"text": "Sample document text.", "metadata": {"author": "John Doe", "category": "example"}}
+    ```
+
+    ## Response Example:
+    ```json
+    {
+        "doc_id": "123456",
+        "text": "Sample document text.",
+        "metadata": {"author": "John Doe"},
+        "is_truncated": true
+    },
+    ```
+    """,
+)
+async def update_document_in_index(
+    index_name: str,
+    doc_id: str,
+    request: UpdateDocumentRequest,
+):
+    try:
+        decoded_index_name = unquote(index_name)
+        decoded_doc_id = unquote(doc_id)
+        await rag_ops.update_document(
+            index_name=decoded_index_name,
+            doc_id=decoded_doc_id,
+            new_doc=request
+        )
+
+        return request
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=str(e))
```

Comment:
Likely updating one or more documents at a time so something like

```suggestion
@app.put("/indexes/{index_name}/documents", response_model=Dict[str, List[str]])
async def update_documents(index_name: str, request: UpdateDocumentsRequest):
    """Batch update documents in an index."""
    try:
        return await rag_ops.update(index_name, request.documents)
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))
```

FastAPI automatically decodes path parameters so don't need unquote

Reply:
updated!